digdag-core/src/main/java/io/digdag/core/storage/ArchiveManager.java

```diff
@@ -99,9 +114,17 @@ private String formatFilePath(int siteId, String projectName, String revisionNam
     {
         Preconditions.checkArgument(!Strings.isNullOrEmpty(projectName), "projectName");
         Preconditions.checkArgument(!Strings.isNullOrEmpty(revisionName), "revisionName");
+        // It's possible the storage used to store a project archive in doesn't accept some characters.
+        // For instance, AWS S3 says the following characters are generally safe
+        //   - Alphanumeric characters [0-9a-zA-Z]
+        //   - Special characters !, -, _, ., *, ', (, and )
+        //
+        // So we'd better encode a project name into the characters above
+        String encodedProjectName = Base64.getEncoder().encodeToString(projectName.getBytes(UTF_8)).replace("=", "_");
         return String.format(ENGLISH,
-                "%d/%s/%s.%s.tar.gz",
-                siteId, projectName, revisionName, DATE_TIME_SUFFIX_FORMAT.format(Instant.now()));
+                "%s%d/%s/%s.%s.tar.gz",
+                pathPrefix, siteId, encodedProjectName, revisionName,
+                DATE_TIME_SUFFIX_FORMAT.format(Instant.now()));
```

Comment:
@frsyuki How about just using UUID instead of `Instant.now`?
It seems the key of S3 file doesn't need to include a timestamp in the name since we can remove expired S3 files using `revisions` table in digdag's database or S3's lifecycle feature.